### ISSUE TITLE
Inserting `use` statements automatically

### ISSUE THREAD

**Issue by tomc1998** [rust-lang/rust.vim#184]
So, my vim setup right now is a total clusterfuck of YouCompleteMe, rust.vim, racer, and ALE - I have no idea what's doing what anymore so I'm just gonna post this here.

Having to insert imports manually every time I go through a compile / run cycle is one of my biggest frustrations, and the thing I miss most from other languages. It genuinely costs me a few minutes of my time each big compile!

Whilst i'm aware this kind of heavyweight functionality probably won't be supported in this plugin, what part of the rust toolset would this fall under? If I wanted to implement this myself, where would I look to and what work has already been done for me?

Thanks, sorry for the fairly off-topic issue. Not too sure if this is really applicable as a github issue, still a bit new to using github as anything but hosting for my personal projects so go easy!^^

**Comment by Mange** [comment]
I think that would fall under the Rust Language Server (RLS).

**Comment by da-x** [comment]
You are completely right, and it's not off topic :) The README.md of rust.vim should be kept updated with best resources available so that new Rust users on Vim will have a good start. It's not the case now. There is going to be a README.md overhaul, so this will be open in the meanwhile.
